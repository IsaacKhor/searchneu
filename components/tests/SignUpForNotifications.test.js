/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */

import React from 'react';

import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import SignUpForNotifications from '../SignUpForNotifications';
import mockData from '../panels/tests/mockData';

Enzyme.configure({ adapter: new Adapter() });

beforeEach(() => {
  window.location.hash = '#fbtest';
});

afterEach(() => {
  window.location.hash = '#';
});

it('should render', () => {
  const result = shallow(
    <SignUpForNotifications
      course={mockData.cs1210}
      userIsWatchingClass={false}
    />
  ).debug();
  expect(result).toMatchSnapshot();
});

it('should render', () => {
  const result = shallow(
    <SignUpForNotifications course={mockData.cs1210} userIsWatchingClass />
  ).debug();
  expect(result).toMatchSnapshot();
});

it('should render the fb button after the button is clicked', async (done) => {
  const wrapper = shallow(
    <SignUpForNotifications
      course={mockData.cs1210}
      userIsWatchingClass={false}
    />
  );
  const instance = wrapper.instance();

  await instance.onSubscribeToggleChange();

  wrapper.update();

  expect(wrapper.contains('Click this button to continue')).toBeTruthy();
  done();
});
