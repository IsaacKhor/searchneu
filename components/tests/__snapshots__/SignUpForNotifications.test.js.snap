// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
"<div className=\\"sign-up-for-notifications-container\\">
  <div className=\\"initialNotificationButton\\" role=\\"button\\" focusable=\\"true\\" tabIndex={0} onClick={[Function: bound onSubscribeToggleChange]}>
    <LogoInput height=\\"14\\" width=\\"18\\" fill=\\"#d41b2c\\" />
    <span>
      Get notified when seats open up!
    </span>
  </div>
  <Modal header=\\"Please disable adblock and sign into Facebook.\\" open={false} content=\\"Please disable any ad blocking extentions for this site because this feature does not work when adblock is enabled. If you are using Firefox in strict blocking mode, you will need to add an exception for this site for this feature to work. You will also have to uninstall Facebook Container for Firefox, if you have that installed. You can also try using a different browser. If you can\\\\'t get it working send us a message at hey@searchneu.com.\\" actions={{...}} centered={true} dimmer={true} closeOnDimmerClick={true} closeOnDocumentClick={false} eventPool=\\"Modal\\" />
</div>"
`;

exports[`should render 2`] = `
"<div className=\\"sign-up-for-notifications-container\\">
  <div className=\\"toggleCTA\\">
    <span>
      Toggle the sections you want to be notified for!
    </span>
  </div>
  <Modal header=\\"Please disable adblock and sign into Facebook.\\" open={false} content=\\"Please disable any ad blocking extentions for this site because this feature does not work when adblock is enabled. If you are using Firefox in strict blocking mode, you will need to add an exception for this site for this feature to work. You will also have to uninstall Facebook Container for Firefox, if you have that installed. You can also try using a different browser. If you can\\\\'t get it working send us a message at hey@searchneu.com.\\" actions={{...}} centered={true} dimmer={true} closeOnDimmerClick={true} closeOnDocumentClick={false} eventPool=\\"Modal\\" />
</div>"
`;
